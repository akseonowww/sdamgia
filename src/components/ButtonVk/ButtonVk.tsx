import React, { ReactNode } from 'react'
import { cn } from '@bem-react/classname'
import { classnames as cx } from '@bem-react/classnames'

import './ButtonVk.scss'

interface IButtonVkProps {
  className?: string
  url: string
  children: ReactNode
}

const ButtonVk = ({ className, url, children }: IButtonVkProps) => {
  const buttonVk = cn('ButtonVk')

  return (
    <a
      className={cx(buttonVk(), className)}
      href={url}
      rel="noopener noreferrer"
    >
      {children}
    </a>
  )
}

export default ButtonVk
