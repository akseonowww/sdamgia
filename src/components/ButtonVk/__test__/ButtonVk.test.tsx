import React from 'react'
import { unmountComponentAtNode, render } from 'react-dom'
import { act } from 'react-dom/test-utils'
import pretty from 'pretty'

import ButtonVk from '../ButtonVk'

describe('ButtonVk', () => {
  let container: HTMLElement | null = null

  beforeEach(() => {
    container = document.createElement('div')
    document.body.appendChild(container)
  })

  afterEach(() => {
    if (container) {
      unmountComponentAtNode(container)
      container.remove()
      container = null
    }
  })

  it('renders with or without a className', () => {
    act(() => {
      render(
        <ButtonVk url="https://vk.com/reshuege">Our group</ButtonVk>,
        container
      )
    })
    expect(container && container.textContent).toBe('Our group')

    act(() => {
      render(
        <ButtonVk className="Student-ButtonVk" url="https://vk.com/reshuege">
          Our group
        </ButtonVk>,
        container
      )
    })
    expect(container && container.textContent).toBe('Our group')
  })

  it('should render a content', () => {
    act(() => {
      render(
        <ButtonVk url="https://vk.com/reshuege">Our group</ButtonVk>,
        container
      )
    })
    if (container) expect(pretty(container.innerHTML)).toMatchSnapshot()

    act(() => {
      render(
        <ButtonVk className="Student-ButtonVk" url="https://vk.com/reshuege">
          Our group
        </ButtonVk>,
        container
      )
    })
    if (container) expect(pretty(container.innerHTML)).toMatchSnapshot()
  })
})
