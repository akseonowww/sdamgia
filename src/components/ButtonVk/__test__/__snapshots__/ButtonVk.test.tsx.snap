// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonVk should render a content 1`] = `"<a class=\\"ButtonVk\\" href=\\"https://vk.com/reshuege\\" rel=\\"noopener noreferrer\\">Our group</a>"`;

exports[`ButtonVk should render a content 2`] = `"<a class=\\"ButtonVk Student-ButtonVk\\" href=\\"https://vk.com/reshuege\\" rel=\\"noopener noreferrer\\">Our group</a>"`;
